javac-plugin/src/main/java/xyz/wagyourtail/jvmdg/javac/JvmdgJavacPlugin.java

```diff
@@ -0,0 +1,170 @@
+import java.io.File;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+
+@SuppressWarnings("UrlHashCode")
+public class JvmdgJavacPlugin implements Plugin, Closeable {
+    private BasicJavacTask task;
+    private Flags flags;
+    private final Set<URL> classpathURLs = new HashSet<>();
+
+    @Override
+    public String getName() {
+        return "jvmdg";
+    }
+
+    static {
+        int vmVersion = Utils.classVersionToMajorVersion(Utils.getCurrentClassVersion());
+        if(vmVersion > 8) {
+            try {
+                Method getModule = Class.class.getDeclaredMethod("getModule");
+                openModule(getModule.invoke(JavacTask.class));
+            } catch (Throwable t) {
+                Utils.sneakyThrow(t);
+            }
+        }
+    }
+
+    @Override
+    public void init(JavacTask t, String... args) {
+        this.task = (BasicJavacTask) t;
+
+        JavaCompiler compiler = JavaCompiler.instance(task.getContext());
+        compiler.closeables = compiler.closeables.prepend(this);
+
+        this.flags = new Flags();
+        flags.api = flags.findJavaApi();
+
+        final String p = "[" + File.separator + "]";
+
+        for(String arg : args) {
+            if(arg.contains("=")) {
+                String[] split = arg.split("=", 2);
+                switch(split[0]) {
+                    case "api":
+                        if(flags.api == null) {
+                            flags.api = new ArrayList<>();
+                        }
+                        for(String s : split[1].split(p)) {
+                            flags.api.add(new File(s));
+                        }
+                        break;
+                    case "logLevel":
+                    case "log":
+                        flags.logLevel = Logger.Level.valueOf(split[1].toUpperCase(Locale.ROOT));
+                        break;
+                    case "skipStubs":
+                        for(String s : split[1].split(",")) {
+                            flags.debugSkipStubs.add(Integer.parseInt(s));
+                        }
+                        break;
+                    case "target":
+                        flags.classVersion = Utils.majorVersionToClassVersion(Integer.parseInt(split[1]));
+                        break;
+                    case "classpath":
+                    case "cp":
+                        for(String s : split[1].split(p)) {
+                            try {
+                                classpathURLs.add(new File(s).toURI().toURL());
+                            } catch (Throwable t1) {
+                                Utils.sneakyThrow(t1);
+                            }
+                        }
+                        break;
+                }
+            }
```

Comment:
does this have to have it's own seperate args. also the args should *at least* match the names of the ones in https://github.com/unimined/JvmDowngrader/blob/main/src/main/java/xyz/wagyourtail/jvmdg/cli/Main.java

also, does it really make sense for the classpath for jvmdg to be different than the classpath provided to javac?

also, also, I believe you can pass VM properties into javac somehow, which can also be used to set the jvmdg flags.

Reply:
I could make it delegate to `Main`, that would just require a little moving of logic in `Main` around so I can get the parsed `Flags`. I'm not a particularly big fan of using `-J` though, I think having the arguments as the arguments to the plugin would work better.